content/opt/setup/30-locales

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+if [[ ! -f /etc/locale.gen.dist ]]
+then
+   cp /etc/locale.gen /etc/locale.gen.dist
+fi
+
+echo "BUILD_LOCALES: $BUILD_LOCALES"
+typeset -i m=$(echo $BUILD_LOCALES | awk -F ',' '{print NF}')
+typeset -i n=1
+while [[ $n -le $m ]]
+do
+   b_locale=$(echo $BUILD_LOCALES | awk -F ',' "{print \$$n}")
+   cat /etc/locale.gen |grep -q "^$b_locale"
+   if [[ $? -ne 0 ]]
+   then
+      echo "adding locale $b_locale"
+      cat /etc/locale.gen | sed "s/# $b_locale/$b_locale/" > /tmp/locale.gen
+      mv /tmp/locale.gen /etc/locale.gen
+   fi
+
+   n=$n+1
+done
```

Comment:
Not further tested, but this would basically do the same:

```
echo "${BUILD_LOCALES}" | while IFS=, read locale charset; do
    sed -i "s/#\\s*${locale}/${locale}/" /etc/locale.gen
done
```